consensus/polybft/consensus_runtime.go

```diff
@@ -646,9 +650,12 @@ func (c *consensusRuntime) deliverMessage(msg *TransportMessage) (bool, error) {
 	return true, nil
 }
 
-func (c *consensusRuntime) runCheckpoint(epoch *epochMetadata) error {
-	// TODO: Implement checkpoint
-	return nil
+// isCheckpointBlock returns indication whether given block is the checkpoint block.
+// Returns true for either epoch ending block or
+// at each N blocks, where N is calculated as division between predefined checkpoint interval and block time
+func (c *consensusRuntime) isCheckpointBlock(blockNumber uint64) bool {
```

Comment:
I would move this as well to `checkpointManager`.

Reply:
The question is what to do with isEndOfEpoch function? It is not needed only for checkpointing as you know...
`func (c *consensusRuntime) isEndOfEpoch(blockNumber uint64) bool`

Are you suggesting splitting isCheckpointBlock into two logical conditions (`isEndOfEpoch || isCheckpointBlock`)?

Reply:
Fixed [6a2a52b](https://github.com/0xPolygon/polygon-edge/pull/861/commits/6a2a52ba4577cd8c48852e9273e77ae3fc5492fe)

Reply:
`isEndOfEpoch` just calls `isEndOfPeriod` function (which is not a method). You can call it from `checkpointManager`, just pass the epoch size to `checkpointManager.isCheckpointBlock` method.

Reply:
This has changed per what we were talking this morning:
epoch ending blocks are determined by comparing epoch numbers in the checkpoint manager [732dd70](https://github.com/0xPolygon/polygon-edge/pull/861/commits/732dd70ad34721da848f89c8016ac39031ac3405)